lib/win32/certstore.rb

```diff
@@ -54,6 +54,12 @@ def add(cert_file_path)
       return add
     end
 
+    def delete(certificate_name)
+      delete_cert = cert_delete(@certstore_handler, certificate_name)
+      close
+      return delete_cert

```

Comment:
I realize this is the existing pattern, but we don't need an explicit return here when this is the last line of the method, i.e. this could just be `delete_cert`.

Similarly, I don't know that we want to close as part of every action. We could update this pattern so this entire method would just be `cert_delete(@certstore_handler, certificate_name)`.

Reply:
Hi @btm, @NimishaS and @mwrock - Please correct me if I'm wrong, as per our discussion in the previous PR: https://github.com/chef/win32-certstore/pull/4#discussion_r135190353, we discided to make `certstore_handler` secure and not accessible outside the class. That's why we created close method as private, having only read permission. For each operation of windows certificate store, we have to create a `certificate_handler` pointer which needs to be destroyed at the end of request. `close` method destroys the pointer.
So if we want to list certificates and add certificate, we have to perform the operations separately in two different blocks and with 2 different certstore_handlers. We can't perform two operations on certificate store with one certstore_handler.